packages/skygear-core/lib/container.js

```diff
@@ -164,15 +164,17 @@ export class BaseContainer {
    * @return {Promise<Object>} promise with result of the lambda function
    */
   lambda(name, data) {
-    return this.publicDB._presave(data)
-      .then((presavedData) => {
-        return this.makeRequest(name, {
-          args: presavedData ? toJSON(presavedData) : undefined
-        });
-      })
-      .then((resp) => {
-        return fromJSON(resp.result);
+    return this.publicDB._presave(
+      this.publicDB._presaveSingleValue.bind(this),
```

Comment:
I found when I put a new Asset (not uploaded) to lambda, I got following exception

```
TypeError: Cannot read property 'makeRequest' of undefined
    at database.js:651
    at new Promise (<anonymous>)
    at makeUploadAssetRequest (database.js:650)
    at Container._presaveSingleValue (database.js:197)
    at PublicDatabase._presave (database.js:255)
    at PublicDatabase._presaveKeyValue (database.js:214)
    at database.js:259
    at arrayMap (lodash.js:660)
    at Function.map (lodash.js:9571)
    at interceptor (lodash.js:16970)
```

Maybe we should bind `this.publicDB` (Database) here instead if `this` (Container) here?

Reply:
My client side testing code
```
const picture = new skygear.Asset({
          name: 'test.png',            // filename of your asset
          base64: 'iVBORw0KGgoAAAANSUhEUgAAAAIAAAABCAYAAAD0In+KAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAAyRpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMy1jMDExIDY2LjE0NTY2MSwgMjAxMi8wMi8wNi0xNDo1NjoyNyAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvIiB4bWxuczp4bXBNTT0iaHR0cDovL25zLmFkb2JlLmNvbS94YXAvMS4wL21tLyIgeG1sbnM6c3RSZWY9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9zVHlwZS9SZXNvdXJjZVJlZiMiIHhtcDpDcmVhdG9yVG9vbD0iQWRvYmUgUGhvdG9zaG9wIENTNiAoTWFjaW50b3NoKSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDo0QjlCMTQ3OTEyRTcxMUU4OEVBNkM1QTRBQzJGNTc1QiIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDo0QjlCMTQ3QTEyRTcxMUU4OEVBNkM1QTRBQzJGNTc1QiI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOjRCOUIxNDc3MTJFNzExRTg4RUE2QzVBNEFDMkY1NzVCIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjRCOUIxNDc4MTJFNzExRTg4RUE2QzVBNEFDMkY1NzVCIi8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+0k1WXQAAABJJREFUeNpiYGBg+A8EDAABBgAN+wP9QoT8VAAAAABJRU5ErkJggg==', // base64 of the file, no mime
          contentType: 'image/png'      // mime of the file
      });
      return skygear.lambda(
        'test_asset', {'asset': picture, 'date': new Date()}
      ).then(response => {
        debugger;
        console.log(response);
      }, function (error) {
        debugger;
        console.log(error);
      });
```

Reply:
> Maybe we should bind this.publicDB (Database) here instead if this (Container) here?
 
right thanks